### PR TITLE
fix(NavExpandable): converts nav expandable to button

### LINKED ISSUE #4377: NavExpandable should be a button and not a link
**Describe the issue. What is the expected and unexpected behavior?**
`NavExpandable` results in a link with `href="#"`.  Since this item is not linkable and will always have an `href` value `#`, it should not be a link.

Expected behavior to be similar to `ExpandableSection` which uses a `button`.

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**

See example https://www.patternfly.org/v4/documentation/react/components/nav#expandable

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
Bugish?

**What is your product and what release version are you targeting?**
OpenShift Console 4.6

### PR DESCRIPTION
**Pull request by jenny-s51**
**What**: Closes #4377.

**Additional issues**: Extends https://github.com/patternfly/patternfly/pull/3250 and includes version bump to pull in those  CSS changes.

### CODE REVIEW

File: packages/react-core/src/components/Nav/NavExpandable.tsx [hunk 1]
```diff
@@ -90,9 +93,27 @@ export class NavExpandable extends React.Component<NavExpandableProps, NavExpand
   };
 
   render() {
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    const { id, title, srText, children, className, isActive, groupId, isExpanded, onExpand, ...props } = this.props;
+    const {
+      onToggle: onToggleProp,
```
Comment (tlabaj) [inline packages/react-core/src/components/Nav/NavExpandable.tsx:97]: You do not need to rename this here.  I would change the name of the function on line 113 instead.  Maybe onClick?  Then you can also get rid of line 111 and declare the function on line 113 as a const.

Comment (jenny-s51) [reply packages/react-core/src/components/Nav/NavExpandable.tsx:97]: good point! will fix.

File: packages/react-core/src/components/Nav/NavExpandable.tsx [hunk 2]
```diff
@@ -92,8 +95,13 @@ export class NavExpandable extends React.Component<NavExpandableProps, NavExpand
   render() {
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const { id, title, srText, children, className, isActive, groupId, isExpanded, onExpand, ...props } = this.props;
+
     const { expandedState } = this.state;
 
+    const onClick = () => {
+      this.setState(prevState => ({ expandedState: !prevState.expandedState }));
```
Comment (tlabaj) [inline packages/react-core/src/components/Nav/NavExpandable.tsx:102]: You need to call onClick here if it is provided.

Comment (jenny-s51) [reply packages/react-core/src/components/Nav/NavExpandable.tsx:102]: What do you mean?

File: packages/react-core/src/components/Nav/NavExpandable.tsx [hunk 3]
```diff
@@ -27,6 +27,8 @@ export interface NavExpandableProps
   id?: string;
   /** allow consumer to optionally override this callback and manage expand state externally */
   onExpand?: (e: React.MouseEvent<HTMLLIElement, MouseEvent>, val: boolean) => void;
+  /** Callback function to toggle the expandable content */
+  onClick?: () => void;
```
Comment (dlabrecq) [inline packages/react-core/src/components/Nav/NavExpandable.tsx:31]: I don't see where this `onClick` prop is called?

File: packages/react-integration/demo-app-ts/src/components/demos/NavDemo/NavDemo.tsx [hunk 4]
```diff
@@ -59,7 +59,7 @@ export class NavDemo extends Component {
         <Title headingLevel="h2" size="2xl">
           Default Nav
         </Title>
-        <div className="example" style={{ border: '1px solid rgb(114, 118, 123)' }}>
+        <div className="example" style={{ backgroundColor: '#212427', border: '1px solid rgb(114, 118, 123)' }}>
```
Comment (tlabaj) [inline packages/react-integration/demo-app-ts/src/components/demos/NavDemo/NavDemo.tsx:62]: why do we need this change?

File: packages/react-integration/demo-app-ts/src/components/demos/NavDemo/NavDemo.tsx [hunk 5]
```diff
@@ -124,7 +124,7 @@ export class NavDemo extends Component {
         <Title headingLevel="h2" size="2xl">
           Expandable Nav
         </Title>
-        <div className="example" style={{ border: '1px solid rgb(114, 118, 123)' }}>
+        <div className="example" style={{ backgroundColor: '#212427', border: '1px solid rgb(114, 118, 123)' }}>
```
Comment (tlabaj) [inline packages/react-integration/demo-app-ts/src/components/demos/NavDemo/NavDemo.tsx:127]: same question for this.

File: packages/react-integration/demo-app-ts/src/components/demos/NavDemo/NavDemo.tsx (hunk 4 above)
Comment (jenny-s51) [reply packages/react-integration/demo-app-ts/src/components/demos/NavDemo/NavDemo.tsx:62]: @tlabaj the nav demo looked like this before adding that styling: https://github.com/patternfly/patternfly/issues/3279. per @redallen, that is actually our default nav styling! He mentioned it needs to be on a dark background for it to look right.

File: packages/react-integration/demo-app-ts/src/components/demos/NavDemo/NavDemo.tsx (hunk 5 above)
Comment (jenny-s51) [reply packages/react-integration/demo-app-ts/src/components/demos/NavDemo/NavDemo.tsx:127]: see above 🙂

**Pull request merged**